### ISSUE TITLE
[Bug]: Condition of Filter errors

### ISSUE THREAD

**Issue by iamlinjunhong** [matrixorigin/matrixone#2913]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

![image](https://user-images.githubusercontent.com/49111204/173034642-101a6560-2e84-4fa9-ac42-399267440872.png)

**Comment by m-schen** [comment]
pr #2914 fix a restrict operator error.

**Comment by m-schen** [comment]
after pr #2896 merged, can be verified.

**Comment by jianwan0214** [comment]
this issue is fixed after pr #2896
